src/core/server/app/middleware/tenant.ts

```diff
@@ -1,16 +1,78 @@
+import { Db } from "mongodb";
 import { v1 as uuid } from "uuid";
 
 import { TenantNotFoundError } from "coral-server/errors";
 import logger from "coral-server/logger";
+import { retrieveSite, Site } from "coral-server/models/site";
+import { retrieveStory } from "coral-server/models/story";
+import { Tenant } from "coral-server/models/tenant";
+import { findSiteByURL } from "coral-server/services/sites";
 import { TenantCache } from "coral-server/services/tenant/cache";
-import { RequestHandler } from "coral-server/types/express";
+import { Request, RequestHandler } from "coral-server/types/express";
+
+interface RequestQuery {
+  parentUrl?: string;
+  storyURL?: string;
+  storyID?: string;
+  siteID?: string;
+}
+
+async function retrieveSiteFromEmbed(
+  mongo: Db,
+  tenant: Tenant,
+  req: Request
+): Promise<Site | null> {
+  // Attempt to detect the site based on the query parameters.
+  const {
+    storyURL = "",
+    storyID = "",
+    parentUrl = "",
+    siteID = "",
+  }: RequestQuery = req.query;
```

Comment:
So there's currently a problem with how we're doing this. The requests where the site is important to the context is ones that request the `viewer`, but because these requests aren't the ones with the iframe parameters on, it means that these will not have the site attached to them, and can therefore never react to a site specific ban.